### PR TITLE
[PyAPI] Add Identity-16 Operation

### PR DESCRIPTION
**Pull request by PiotrKrzem**
### Details:
 - Extends Python API with Identity operation 
 - Identity op Core/Ref files should disappear after merging: https://github.com/openvinotoolkit/openvino/pull/26716

### Tickets:
 - 152728

### CODE REVIEW

File: src/bindings/python/src/openvino/runtime/opset15/ops.py [hunk 1]
```diff
@@ -251,6 +251,24 @@ def bitwise_left_shift(
...
+@nameable_op
+def identity(
+    data: NodeInput,
+    name: Optional[str] = None,
+) -> Node:
+    """Identity operation is used as a placeholder. It creates a copy of the input to forward to the output.
+
+    :param data: Tensor with data.
+
+    :return: The new node performing BitwiseRightShift operation.
```
Comment (p-wysocki) [inline src/bindings/python/src/openvino/runtime/opset15/ops.py:263]:
```suggestion
    :return: The new node performing Identity operation.
```

File: src/core/reference/include/openvino/reference/identity.hpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+#include <cstring>
+
+namespace ov {
+namespace reference {
+/**
+ * @brief Identity operation computes the identity of the input tensor.
+ *
+ * @param input Input matrix (matrices) pointer.
+ * @param output Output matrix (matrices) pointer.
```
Comment (praasz) [inline src/core/reference/include/openvino/reference/identity.hpp:15]: Detail: all parameters should be documented.

File: src/core/src/op/identity.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    OV_OP_SCOPE(v15_Identity_visit_attributes);
+    return true;
+}
+
+void Identity::Identity::validate_and_infer_types() {
+    OV_OP_SCOPE(v15_Identity_validate_and_infer_types);
+
+    const auto input_shapes = ov::util::get_node_input_partial_shapes(*this);
+
+    set_output_type(0, get_input_element_type(0), input_shapes[0]);
```
Comment (praasz) [inline src/core/src/op/identity.cpp:33]:
```suggestion
    set_output_type(0, get_input_element_type(0), get_input_partial_shape(0));
```

File: src/core/tests/type_prop/identity.cpp [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+#include "openvino/op/constant.hpp"
+
+using namespace testing;
+
+namespace ov {
+namespace test {
+
+class TypePropIdentityV15Test : public TypePropOpTest<op::v15::Identity> {};
+
+TEST_F(TypePropIdentityV15Test, default_ctor) {
```
Comment (praasz) [inline src/core/tests/type_prop/identity.cpp:20]: Add some test when standard constructors are used?

File: src/plugins/template/backend/ops/identity.cpp [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    const auto input_shape = inputs[0].get_shape();
+    const auto total_elements =
+        std::accumulate(input_shape.begin(), input_shape.end(), static_cast<size_t>(1), std::multiplies<size_t>());
+    const auto total_size_in_bytes = get_memory_size(inputs[0].get_element_type(), total_elements);
+
+    outputs[0].set_shape(input_shape);
+
+    ov::reference::identity(static_cast<const char*>(inputs[0].data()),
+                            static_cast<char*>(outputs[0].data()),
+                            total_size_in_bytes);
```
Comment (praasz) [inline src/plugins/template/backend/ops/identity.cpp:23]:
```suggestion
    ov::reference::identity(static_cast<const char*>(inputs[0].data()),
                            static_cast<char*>(outputs[0].data()),
                            inputs[0].get_byte_size());
```

Comment (PiotrKrzem) [reply src/plugins/template/backend/ops/identity.cpp]: Applied in Ref/Core!

File: src/core/src/op/identity.cpp (hunk 3 above)
Comment (PiotrKrzem) [reply src/core/src/op/identity.cpp]: Applied in Ref/Core!

File: src/core/reference/include/openvino/reference/identity.hpp (hunk 2 above)
Comment (PiotrKrzem) [reply src/core/reference/include/openvino/reference/identity.hpp:15]: Applied in Ref/Core!

File: src/bindings/python/src/openvino/runtime/opset15/ops.py (hunk 1 above)
Comment (PiotrKrzem) [reply src/bindings/python/src/openvino/runtime/opset15/ops.py]: Applied, great catch!

**Pull request merged**